### PR TITLE
Creates /var/lib/kubelet as root

### PR DESCRIPTION
**Pull request by alindeman**
**What this PR does / why we need it**: `hack/local-cluster-up.sh` does not currently use sudo to create `/var/lib/kubelet`, so it fails unless the entire script is run with sudo.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: hack/local-up-cluster.sh [hunk 1]
```diff
@@ -595,7 +595,7 @@ function start_kubelet {
       priv_arg="--allow-privileged "
     fi
 
-    mkdir -p /var/lib/kubelet
+    sudo mkdir -p /var/lib/kubelet
```
Comment (dims) [inline hack/local-up-cluster.sh]: @alindeman in line 247, the CERT_DIR is created using:
`mkdir -p "${CERT_DIR}" &>/dev/null || sudo mkdir -p "${CERT_DIR}"`

Do you want to use the same pattern here?
`mkdir -p /var/lib/kubelet &>/dev/null || sudo mkdir -p /var/lib/kubelet`

Comment (alindeman) [inline hack/local-up-cluster.sh]: :+1: I like it. I've updated the diff with that.

**Comment by cblecker** [comment]
Creating this dir as root creates another set of issues as the dir won't be writable by the current user if it's created as root. The `${CERT_DIR}` is it's own beast looking at the script because if you require sudo, then every command that references it must also be sudo'd so that you have permissions.

An alternative to this would be that if mkdir errors, then print a warning to create it yourself, and then abort. Or, perhaps make it configurable so that the kubelet dir can be placed somewhere that the current user has permissions to create.

**Comment by alindeman** [comment]
> Creating this dir as root creates another set of issues as the dir won't be writable by the current user if it's created as root. The ${CERT_DIR} is it's own beast looking at the script because if you require sudo, then every command that references it must also be sudo'd so that you have permissions.

@cblecker `kubelet` is already run with `sudo` and is the only place we use `/var/lib/kubelet` and $POD_MANIFEST_PATH as far as I can tell. Could you help me understand more specifically where the concern about creating the directory as root is?

**Comment by cblecker** [comment]
if `DOCKERIZE_KUBELET` is enabled, docker doesn't run with sudo and tries to mount this dir into the container.

**Comment by alindeman** [comment]
> if DOCKERIZE_KUBELET is enabled, docker doesn't run with sudo and tries to mount this dir into the container.

Indeed, but `docker run` is just a way to make an API request to the docker daemon, which does run as root and would be the thing to actually interact with `/var/lib/kubelet`. And `kubelet` would run as root inside the container, right?

Maybe if you could point me to a specific way this will fail, it would be easier for me to know how to proceed.

File: hack/local-up-cluster.sh [hunk 2]
```diff
@@ -588,14 +588,14 @@ function start_controller_manager {
 
 function start_kubelet {
     KUBELET_LOG=${LOG_DIR}/kubelet.log
-    mkdir -p ${POD_MANIFEST_PATH} || true
+    mkdir -p ${POD_MANIFEST_PATH} &>/dev/null || sudo mkdir -p ${POD_MANIFEST_PATH} || true
```
Comment (cblecker) [inline hack/local-up-cluster.sh]: This doesn't need `|| true` at the end anymore if we are going this route.

**Review by cblecker — COMMENTED** [review]
I just poked at the script again and it looks like the `DOCKERIZE_KUBELET` code path may not even be working at all, so my concerns may be moot... and the direct route would run the kubelet with sudo anyways.

Okay then.. one comment, and we can run this through CI :)

**Comment by alindeman** [comment]
@cblecker Thanks! I've updated the commit based on your feedback.

**Comment by alindeman** [comment]
> W0814 22:09:00.687]  - Quota 'SUBNETWORKS' exceeded.  Limit: 150.0

/retest

**Comment by cblecker** [comment]
/approve
LGTM to @dims :)

**Comment by alindeman** [comment]
@dims Do you have time to take a look at this one?

@cblecker I might need an 'approve no-issue' unless you think I need to file an issue about it.

**Comment by dims** [comment]
/lgtm

(not sure if my "/approve noissue" will work)

**Pull request merged**